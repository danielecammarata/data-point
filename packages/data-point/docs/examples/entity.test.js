/* eslint-env jest */

// eslint-disable-next-line import/no-extraneous-dependencies
const DataPoint = require("data-point");

const checkCardInputSchema = DataPoint.createTypeCheckReducer(input => {
  return typeof input.uid === "number" && typeof input.shortTitle === "string";
}, "{uid:Number, shortTitle:String}");

const CardModel = DataPoint.Model("Card", {
  inputType: checkCardInputSchema,
  value: {
    id: "$uid",
    title: "$shortTitle"
  }
});

describe("Card Entity", () => {
  it("should create Card Object", async () => {
    const dataPoint = DataPoint.create();
    const input = {
      uid: 123,
      shortTitle: "Some Title"
    };

    await expect(dataPoint.resolve(CardModel, input)).resolves.toEqual({
      id: 123,
      title: "Some Title"
    });
  });

  it("should check for input type", async () => {
    const dataPoint = DataPoint.create();

    const input = {
      shortTitle: "Some Title"
    };

    await expect(dataPoint.resolve(CardModel, input)).rejects.toMatchSnapshot();
  });
});
