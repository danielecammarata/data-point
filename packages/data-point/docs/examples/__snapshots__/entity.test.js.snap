// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Entity should check for input type 1`] = `
[Error: Entity type check failed!
Expected type: {uid:Number, shortTitle:String}
Actual type: object
Input value: { shortTitle: 'Some Title' }]
`;
